flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitGenerator.java

```diff
@@ -27,23 +27,39 @@
 import org.apache.iceberg.TableProperties;
 import org.apache.iceberg.TableScan;
 import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.flink.source.split.IcebergSourceSplit;
 import org.apache.iceberg.io.CloseableIterable;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
-class FlinkSplitGenerator {
+public class FlinkSplitGenerator {
   private FlinkSplitGenerator() {
   }
 
   static FlinkInputSplit[] createInputSplits(Table table, ScanContext context) {
-    List<CombinedScanTask> tasks = tasks(table, context);
-    FlinkInputSplit[] splits = new FlinkInputSplit[tasks.size()];
-    for (int i = 0; i < tasks.size(); i++) {
-      splits[i] = new FlinkInputSplit(i, tasks.get(i));
+    try (CloseableIterable<CombinedScanTask> tasksIterable = planTasks(table, context)) {
+      List<CombinedScanTask> tasks = Lists.newArrayList(tasksIterable);
+      FlinkInputSplit[] splits = new FlinkInputSplit[tasks.size()];
+      for (int i = 0; i < tasks.size(); i++) {
+        splits[i] = new FlinkInputSplit(i, tasks.get(i));
+      }
+      return splits;
+    } catch (IOException e) {
+      throw new UncheckedIOException("Failed to process tasks iterable", e);
+    }
+  }
+
+  public static List<IcebergSourceSplit> planIcebergSourceSplits(
```

Comment:
Should we add a javadoc to indicate why do we need to add an extra `planIcebergSourceSplits` (compared to `createInputSplits`) ?  I think it's used for implementing the flip27's SourceSplit, right ?

Reply:
It's good to align with the `createInputSplits` by naming this as `createIcebergSourceSplits`.

Reply:
actually, I think we should rename `createInputSplits` to `planFlinkInputSplit`. We are not creating splits out of nowhere. Both are just discover/plan splits from table and call the same `planTasks`. I can add some javadoc on the new `planIcebergSourceSplits` method.

Since `createInputSplits` is non-public, we should be safe to rename.